basxconnect/core/layouts/admin_layout.py

```diff
@@ -57,4 +61,34 @@ def maintainance_package_layout(request):
 
 
 def maintenance_database_optimization(request):
-    return hg.H3("Coming Soon")
+    # get the database's size (in kilobytes)
+    # TODO: Add a button that runs the command
+    current_db_size = os.stat(os.getcwd() + "/db.sqlite3").st_size / 1000
+    previous_size = 0
+
+    optimize_btn = hg.FORM(
+        layout.form.CsrfToken(),
+        Button("Optimize", type="submit"),
+        hg.INPUT(type="hidden", name="previous-size", value=current_db_size),
+        method="POST",
+    )
+
+    if request.method == "POST":
+        cursor = connection.cursor()
+        cursor.execute("VACUUM;")
+        current_db_size = os.stat(os.getcwd() + "/db.sqlite3").st_size / 1000
+        previous_size = request.body.decode("utf-8")
+        previous_size = float(previous_size[previous_size.find("previous-size") + 14 :])
+
+    return hg.BaseElement(
+        hg.If(
+            request.method == "POST",
+            hg.H5(f"Previous Size: {previous_size : .2f} KB"),
+            hg.H5(f"Current Size: {current_db_size : .2f} KB"),
+        ),
+        hg.If(
+            request.method == "POST",
+            hg.H5(f"Minimized Size: {current_db_size : .2f} KB"),
+        ),
```

Comment:
I recommend we use ```django.contrib.messages``` to display the information (see https://docs.djangoproject.com/en/dev/ref/contrib/messages/#adding-a-message bread will automatically display the messages).

Reply:
Kilobytes should be written as "kB" instead of "KB"